lib/systems/default.nix

```diff
@@ -194,6 +194,34 @@ let
          || isWasm                                                     # WASM
         ) && !isStatic;
 
+      # Whether the system provides its own libunwind implementation that we
+      # should use.
+      hasSystemLibunwind = final.isFreeBSD || final.isDarwin;
```

Comment:
I don't think this works since we have multiple unwinder implementations.
- LLVM libunwind
- Darwin libunwind
- non-GNU libunwind
- GCC's unwinder
Instead, I think treating it like an enum like #365057 does works best imo.